src/Store/SearchStoreConnector.re

```diff
@@ -0,0 +1,83 @@
+module Core = Oni_Core;
+module Model = Oni_Model;
+
+open Model.Actions;
+
+let start = () => {
+  let (stream, _dispatch) = Isolinear.Stream.create();
+
+  let searchUpdater = (state: Model.Search.t, action) => {
+    switch (action) {
+    | SearchInput(text, cursorPosition) => {
+        ...state,
+        queryInput: text,
+        cursorPosition,
+      }
+
+    | SearchStart => {...state, query: state.queryInput,hits: []}
+
+    | SearchUpdate(items) => {...state, hits: state.hits @ items}
+
+    // | SearchComplete
+
+    | SearchSelectResult(match) =>
+      print_endline("!! SELECT: " ++ match.file);
```

Comment:
It seems like here is where we could `dispatch(OpenFileByPath(match.file, ...))` and maybe hook up the line number, too!